src/event/player/PlayerEntityInteractEvent.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\event\player;
+
+use pocketmine\entity\Entity;
+use pocketmine\event\Cancellable;
+use pocketmine\event\CancellableTrait;
+use pocketmine\item\Item;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+
+/**
+ * Called when a player interacts with an entity
+ */
+class PlayerEntityInteractEvent extends PlayerEvent implements Cancellable{
+	use CancellableTrait;
+
+	/** @var Entity */
+	protected $entity;
+
+	/** @var Item */
+	protected $item;
+
+	/** @var Vector3 */
+	protected $clickPos;
+
+	/** @var int */
+	protected $slot;
+
+	public function __construct(Player $player, Entity $entity, Item $item, Vector3 $clickPos, int $slot){
```

Comment:
This can use promoted properties since we're targeting PHP 8.0 now.

Reply:
I'm using a quite outdated version of PHPStorm that doesn't fully cover PHP8, but I assume this is what you're referring to?

```php
class PlayerEntityInteractEvent extends PlayerEvent implements Cancellable{
	use CancellableTrait;

	public function __construct(
		protected Player $player,
		protected Entity $entity,
		protected Vector3 $clickPos
	){}

	public function getEntity() : Entity{
		return $this->entity;
	}

	public function getClickPosition() : Vector3{
		return $this->clickPos;
	}
}
```

Reply:
Yes, except it should be private, not protected.

Reply:
the `player` property is inherited from the `PlayerEvent`, so I'm once again assuming that it should work like this:
```php
	public function __construct(
		protected Player $player,
		private Entity $entity,
		private Vector3 $clickPos
	){}
```

Reply:
No, the Player parameter shouldn't be promoted. Pass it to the parent constructor.

Reply:
Like this?

```php
	public function __construct(
		Player $player,
		private Entity $entity,
		private Vector3 $clickPos
	){
		$this->player = $player;
	}
```

Reply:
PlayerEvent has no constructor so it should be right